src/Appwrite/Extend/PDOStatement.php

```diff
@@ -98,7 +98,7 @@ public function fetch($fetch_style = PDONative::FETCH_ASSOC, $cursor_orientation
         return $result;
     }
 
-    public function fetchAll($how = null, $class_name = null, $ctor_args = null)
+    public function fetchAll(int $fetch_style = PDO::FETCH_BOTH, mixed ...$fetch_args)
     {
         $result = $this->PDOStatement->fetchAll();
```

Comment:
Could it be, that you forgot to add the parameters to `fetchAll()`? You may also restore `$ctor_args` and pass them, too.

Reply:
I'll add both return types and doc blocks 👍🏻

Seems like in PHP8 this method signature was changed to: `(int $fetch_style = PDO::FETCH_BOTH, mixed ...$fetch_args)` I only changed it because it was throwing errors.

Reply:
This should work fine:

```php
public function fetchAll(int $fetch_style = PDO::FETCH_BOTH, mixed ...$fetch_args, array $ctor_args = []): array
{
    return $this->PDOStatement->fetchAll($fetch_style, $fetch_args, $ctor_args);
}
```

Reply:
Yes, but this function can also return false. 🤔 I'm also not sure the $ctor_args is still supported as in PHP8 $fetch_args is now using the splat operator.

Reply:
> Yes, but this function can also return false. 🤔 I'm also not sure the $ctor_args is still supported as in PHP8 $fetch_args is now using the splat operator.

Well, I haven't found any information regarding a signature change of this method. And yes, you are right, that it can also return false (seems to be a fault in the official documentation). In this case, this should fit:

```php
public function fetchAll(int $fetch_style = PDO::FETCH_BOTH, mixed ...$fetch_args): mixed
{
    return $this->PDOStatement->fetchAll($fetch_style, $fetch_args);
}
```

Reply:
With proper type hinting, the doc block becomes obsolete :P